### PR TITLE
Update switch.py

### PR DESCRIPTION
**Pull request by Bullischorsch**
Changes based on latest version:

1. correction of checking and setting values of "mode" and "value"
2. changed the icons to "cog" in different states that show current operation mode:
  - filled icons, if switch is the driving one (automated in mode AUTO, manual in mode HAND), otherwise outlined
  - cog if current mode is "EIN", cog-off if current mode is "AUS"

### CODE REVIEW

**Comment by nielstron** [comment]
Hi @Bullischorsch,
thanks for your proposals! I have made my own adjustments to the code, I will post detailed reasons at #3

**Comment by nielstron** [comment]
Do you consider the attribute `fullmode` useful and meaningful to include? I found that it is rather redundant, given the state of the switch and the `mode` attribute.

File: custom_components/blnet/switch.py [hunk 1]
```diff
@@ -114,15 +126,17 @@ def is_on(self):
 
     def turn_on(self, **kwargs):
         """Turn the device on."""
-        self.communication.turn_on(self._id)
-        self._state = STATE_ON
-        self._assumed_state = True
+        if self._mode != 'AUTO':
```
Comment (nielstron) [inline custom_components/blnet/switch.py:129]: I see your intention. However I decided to make it possible to manually override automatically driven switches. The reason is that waiting for the status update after first turning the switch from "auto" to "hand" to then change the current state too cumbersome.

Comment (Bullischorsch) [reply custom_components/blnet/switch.py:129]: Of cause, that's right. Best would be to have a control with 3 states in HA. I had a look at entity-type input_select, but that's not very comfortable. Maybe I'll try this extention one day:
https://github.com/thomasloven/lovelace-slider-entity-row

Comment (nielstron) [reply custom_components/blnet/switch.py:129]: Yeah that would be my preferred option too... If you get any useful results let me know :)

**Pull request closed without merging**